File: examples/contract_query_state_variable/query_state_variable.rb
require_relative '../../wrappers/Ruby/aergo.rb'

aergo = Aergo.new("testnet-api.aergo.io", 7845)

ret = aergo.query_smart_contract_state_variable(
    "AmhcceopRiU7r3Gwy5tmtkk4Z3Px53SfsKBifGMvaSSNiyWrvKYe",
    "Name")

if ret["success"]
  puts "Smart Contract State Variable Query OK"
  puts "Response: " + ret["result"]
else
  puts "FAILED when querying the smart contract state variable"
  puts "Response: " + ret["result"]
end

File: examples/contract_query_state_variable/query_state_variable.cpp
#include "aergo.hpp"

int main() {
  Aergo aergo("testnet-api.aergo.io", 7845);
  char response[1024];

  std::cout << "Connected. Sending request...\n";

  bool success = aergo.query_smart_contract_state_variable(
    response, sizeof response,
    "AmhcceopRiU7r3Gwy5tmtkk4Z3Px53SfsKBifGMvaSSNiyWrvKYe",
    "Name"
  );

  std::cout << "------------------------------------\n";
  std::cout << "Smart Contract State Variable Query " << (success ? "OK" : "FAILED") << "\n";
  std::cout << "Response: " << response << "\n";
  std::cout << "------------------------------------\n";

  std::cout << "Disconnected\n";
  return 0;
}
